lib/foreman_maintain/package_manager/dnf.rb

```diff
@@ -5,6 +5,26 @@ def clean_cache(assumeyes: false)
       super
     end
 
+    def dnf_plugin_dirs
+      dnf_plugin_parent_dirs = Dir.glob('/usr/lib/python3*').map { |d| d + '/site-packages' }
+      dnf_plugin_parent_dirs.map do |d|
+        if Dir.new(d).children.include?('dnf-plugins')
+          d + '/dnf-plugins'
+        end
+      end.compact
+    end
+
+    def protector_plugin_files
+      @protector_plugin_files ||= dnf_plugin_dirs.map do |d|
+        d + '/' + 'foreman-protector.py'
+      end
+    end
+
+    def version_locking_enabled?
+      super &&
+        protector_plugin_files.each { |f| File.exist?(f) }
+    end
```

Comment:
what does "enabled?" mean?
* active? then the check is wrong
* available? then this should just return true as we support locking for dnf.